libmscore/scale.h

```diff
@@ -0,0 +1,50 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2002-2012 Werner Schweer
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2
+//  as published by the Free Software Foundation and appearing in
+//  the file LICENCE.GPL
+//=============================================================================
+
+#ifndef __SCALE_H__
+#define __SCALE_H__
+
+/**
+ \file
+ Definition of Scale class.
+ */
+
+#include "note.h"
+
+namespace Ms {
+
+class Scale {
+public:
+      bool loadScale(const QString&);
+      double getTuning(const Note* note) const;
+
+private:
+      QString getNextLine(QTextStream& in);
+
+      std::vector<double> originalNotes;
+      std::map<Tpc, double> computedNotes;
+
```

Comment:
I think both approaches are good. I choose this one because of the following reasons : 
  - it's easier to see which enum I'm using.
  - the code would not look as good trying to write TPC_C - TPC_MIN because TPC_MIN is not 0 but -1.
  - all values are placed in the vector in a manner easy to understand just by looking at scale.cpp, while storing by the index of the enum it would be hard to figure out which note it is unless you go and check that enum (because it starts with all BB and then B, normal, S, and SS).